fish-rust/src/wcstringutil.rs

```diff
@@ -464,38 +470,34 @@ pub fn trim(input: WString, any_of: Option<&wstr>) -> WString {
 /// \p idx may be "one past the end."
 pub fn count_preceding_backslashes(text: &wstr, idx: usize) -> usize {
     assert!(idx <= text.len(), "Out of bounds");
-    let mut backslashes = 0;
-    while backslashes < idx && text.char_at(idx - backslashes - 1) == '\\' {
-        backslashes += 1;
-    }
-    backslashes
+    text.chars().take(idx).take_while(|&c| c == '\\').count()
```

Comment:
This looks wrong - this looks to be counting backslashes directly after `idx` but it should count the number of backslashes directly before `idx`.